### PR TITLE
Bugfix/image size is reduce to device size

### LINKED ISSUE #11: cropped image can't keep high quality
I don’t know if you find that the picture after cropped is a jagged picture

### PR DESCRIPTION
**Pull request by ikbendewilliam**
Fixes #11

### CODE REVIEW

File: lib/src/widgets/custom_image_crop_widget.dart [hunk 1]
```diff
@@ -20,6 +19,7 @@ class CustomImageCrop extends StatefulWidget {
   final CustomCropShape shape;
   final double cropPercentage;
   final CustomPaint Function(Path) drawPath;
+  late final imagePaintDuringCrop;
```
Comment (NicolaVerbeeck) [inline lib/src/widgets/custom_image_crop_widget.dart:22]: untyped

File: lib/src/widgets/custom_image_crop_widget.dart [hunk 2]
```diff
@@ -37,22 +37,30 @@ class CustomImageCrop extends StatefulWidget {
...
     this.overlayColor = const Color.fromRGBO(0, 0, 0, 0.5),
     this.backgroundColor = Colors.white,
     this.shape = CustomCropShape.Circle,
     this.cropPercentage = 0.8,
     this.drawPath = DottedCropPathPainter.drawPath,
+    Paint? imagePaintDuringCrop,
     Key? key,
-  }) : super(key: key);
+  }) : super(key: key) {
+    if (imagePaintDuringCrop == null) {
```
Comment (NicolaVerbeeck) [inline lib/src/widgets/custom_image_crop_widget.dart:51]: this.imagePaintDuringCrop =  imagePaintDurtingCrop ?? ...

**Review by NicolaVerbeeck — CHANGES_REQUESTED** [review]
Also add a pubignore file that excludes most 'heavy' example stuff (keep only dart code) to reduce load on pub.dev and clients

File: lib/src/widgets/custom_image_crop_widget.dart (hunk 2 above)
Comment (ikbendewilliam) [reply lib/src/widgets/custom_image_crop_widget.dart:51]: Yes, I thought so to, but we do `..filterQuality = FilterQuality.high`
The following code snippet illustrates the issue:
```dart
  final List? list = [];
  final list2 = list ?? []..add('');
  print(list2.length); // 1
```

Comment (NicolaVerbeeck) [reply lib/src/widgets/custom_image_crop_widget.dart:51]: Add parentesis

File: lib/src/widgets/custom_image_crop_widget.dart [hunk 3]
```diff
@@ -37,22 +37,27 @@ class CustomImageCrop extends StatefulWidget {
...
     required this.cropController,
     this.overlayColor = const Color.fromRGBO(0, 0, 0, 0.5),
     this.backgroundColor = Colors.white,
     this.shape = CustomCropShape.Circle,
     this.cropPercentage = 0.8,
     this.drawPath = DottedCropPathPainter.drawPath,
+    Paint? imagePaintDuringCrop,
     Key? key,
-  }) : super(key: key);
+  }) : super(key: key) {
```
Comment (NicolaVerbeeck) [inline lib/src/widgets/custom_image_crop_widget.dart:50]: Could you move this to the constructor and thus remove the late final

Comment (ikbendewilliam) [reply lib/src/widgets/custom_image_crop_widget.dart:50]: You mean like this? `this.imagePaintDuringCrop = const (Paint()..filterQuality = FilterQuality.high),`
> The constructor being called isn't a const constructor.
> Try removing 'const' from the constructor invocation

Or like this? `this.imagePaintDuringCrop = (Paint()..filterQuality = FilterQuality.high),`
> The default value of an optional parameter must be constant

Comment (ikbendewilliam) [reply lib/src/widgets/custom_image_crop_widget.dart:50]: Yes

**Review by NicolaVerbeeck — APPROVED** [review]

**Pull request merged**